gui/rendering/Node.hpp

```diff
@@ -75,23 +83,27 @@ class Node : public std::enable_shared_from_this<Node> {
    * Adds another node as a direct child. The node will assume ownership
    * of the node added as a child.
    * @param node
+   * @param localFrameCoord
    */
-  void add(std::shared_ptr<Node> node, bool transformInLocalSpace = true);
+  void add(std::shared_ptr<Node> node, bool localFrameCoord = true);
 
   /**
-   * Removes the node from its parent tree.
+   * Get the root node of the tree
    */
-  void remove();
+  std::shared_ptr<Node> getRoot();
 
   /**
-   * Traverses the tree and updates child transformations.
+   * Removes the node from its parent tree.
    */
-  void updateWorldTransform(float delta = 1 / 60.0f);
+  void remove();
 
   /**
-   * Recreates the local-space transform based on pos, rot, and scale.
+   * entry point for update the scene
+   * @param delta
    */
-  virtual void updateLocalTransform();
+  void updateSceneGraph(float delta = 0);
```

Comment:
should this be private? it seems like something only the root node should have?

Reply:
If it's private you can't use it anymore, not even from the root node. Am I missing something?

Reply:
I guess the way to do this cleanly would be to have a `RootNode` that inherits `Node`, which can contain things like this that only the root node should have? Otherwise all `Node` objects have this `UpdateSceneGraph` function but I guess it only makes sense to call it on the root node?